### PR TITLE
feat: refactor RPC `starkNet_getTransactions` with `StarkScan` DataClient

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
This PR is refactor RPC `starkNet_getTransactions` to switch the data provider from Voyager to StarkScan  

it includes:
- refactor RPC `starkNet_getTransactions` with new RPC structure.
- apply StarkScan into  `starkNet_getTransactions`.
- add `TransactionService` to encapsulate the logic for list transactions.
- the data provided by the provider will not be stored in the state. Only the transaction sent from the wallet UI will be stored until it is confirmed by the L2 Chain.

### CODE REVIEW

**Review by khanti42 — CHANGES_REQUESTED** [review]

File: packages/starknet-snap/src/chain/transaction-service.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  ): AsyncGenerator<Transaction> {
+    for (const tx of transactions) {
+      // Only return transaction that are related to the contract address or a deployed transactions.
+      if (this.isTransactionHasSameContractOrAnDeploy(tx, contractAddress)) {
+        yield tx;
+      }
+    }
+  }
+
+  protected isTransactionHasSameContractOrAnDeploy(
```
[the remark this answers is not in the document]
Comment (stanleyyconsensys) [reply packages/starknet-snap/src/chain/transaction-service.ts:80]: for boolean, we start with Is or has

File: packages/starknet-snap/src/index.tsx [hunk 2]
```diff
@@ -52,6 +52,7 @@ import {
   watchAsset,
   getAddrFromStarkName,
   getTransactionStatus,
+  ListTransactions,
```
[the remark this answers is not in the document]
Comment (stanleyyconsensys) [reply packages/starknet-snap/src/index.tsx:55]: good catch

File: packages/starknet-snap/src/chain/transaction-service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (stanleyyconsensys) [reply packages/starknet-snap/src/chain/transaction-service.ts:80]: as it is in transaction service, so we know it is for transaction

may be rename it to 
`hasMatchingContractOrIsDeploy`?

wdyt?

[the remark this answers is not in the document]
Comment (khanti42) [reply packages/starknet-snap/src/chain/transaction-service.ts:80]: hasMatchingContractOrIsDeploy works great!

**Review by khanti42 — CHANGES_REQUESTED** [review]

**Review by khanti42 — APPROVED** [review]
LGTM

**Pull request merged**